great_tables/_text.py

```diff
@@ -12,38 +12,65 @@
 class Text:
     text: str
 
+    def to_html(self) -> str:
+        return self.text
 
+    def to_latex(self) -> str:
+        return self.text
+
+
+@dataclass
 class Md(Text):
     """Markdown text"""
 
+    def to_html(self) -> str:
+        return _md_html(self.text)
+
+    def to_latex(self) -> str:
+        return _md_latex(self.text)
+
 
+@dataclass
 class Html(Text):
     """HTML text"""
 
+    def to_html(self) -> str:
+        return self.text
+
+    def to_latex(self) -> str:
+        return self.text
+
 
 def _md_html(x: str) -> str:
     str = commonmark.commonmark(x)
     return re.sub(r"^<p>|</p>\n$", "", str)
 
 
-def _process_text(x: str | Text | None) -> str:
+def _md_latex(x: str) -> str:
+
+    # TODO: Implement commonmark to LaTeX conversion (through a different library as
+    # commonmark-py does not support it)
+    raise NotImplementedError("Markdown to LaTeX conversion is not supported yet")
+
+
+def _process_text(x: str | Text | None, context: str = "html") -> str:
+
     from great_tables._helpers import UnitStr
 
     if x is None:
         return ""
 
-    if isinstance(x, Md):
-        return _md_html(x.text)
-    elif isinstance(x, Html):
-        return x.text
-    elif isinstance(x, str):
-        return _html_escape(x)
-    elif isinstance(x, Text):
-        return x.text
-    elif isinstance(x, UnitStr):
-        return x.to_html()
-    else:
-        raise TypeError(f"Invalid type: {type(x)}")
+    escape_fn = _html_escape if context == "html" else _latex_escape
+
+    if isinstance(x, str):
+
+        return escape_fn(x)
+
+    elif isinstance(x, (Md, Text, Html, UnitStr)):
```

Comment:
From pairing, we could create a parent class for these pieces (e.g. Md and Text inherit from BaseText, etc..)